app/models/team.rb

```diff
@@ -4,14 +4,21 @@ class Team < ActiveRecord::Base
 
   has_many :namespaces
 
-  # Users & owners
   has_many :team_users
   has_many :users, through: :team_users
-  has_many :owners, -> { where 'team_users.owner': true },
+  has_many :owners, -> { where 'team_users.role' => TeamUser.roles['owner'] },
+    through: :team_users, source: :user
+  has_many :contributors, -> { where 'team_users.role' => TeamUser.roles['contributor'] },
+    through: :team_users, source: :user
+  has_many :viewers, -> { where 'team_users.role' => TeamUser.roles['viewer'] },
```

Comment:
Good point, they are useless.

I'm going to remove them.

Anything else? In the meantime thanks
On 29 Apr 2015 18:31, "Miquel Sabaté Solà" <notifications@github.com> wrote:

> In app/models/team.rb
> <https://github.com/SUSE/Portus/pull/29#discussion_r29353574>:
>
> >    has_many :team_users
> >    has_many :users, through: :team_users
> > -  has_many :owners, -> { where 'team_users.owner': true },
> > +  has_many :owners, -> { where 'team_users.role' => TeamUser.roles['owner'] },
> > +    through: :team_users, source: :user
> > +  has_many :contributors, -> { where 'team_users.role' => TeamUser.roles['contributor'] },
> > +    through: :team_users, source: :user
> > +  has_many :viewers, -> { where 'team_users.role' => TeamUser.roles['viewer'] },
>
> I believe that viewers won't be used. That's because any member of the
> team has effectively pull access, so the viewers relationship is redundant
> here (see the NamespacePolicy::pull? method from your PR).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SUSE/Portus/pull/29/files#r29353574>.
>